esp-hal-common/src/soc/mod.rs

```diff
@@ -10,3 +10,42 @@ pub use self::soc::*;
 mod soc;
 
 mod efuse_field;
+
+pub fn is_valid_ram_address(address: u32) -> bool {
+    #[cfg(feature = "esp32")]
+    if !(0x3FFA_E000..=0x4000_0000).contains(&address) {
```

Comment:
Feel free to leave this for another PR (or just ignore it altogether), but maybe it would be good to introduce some new constants in the `soc` module and simplify this function? ESP-IDF uses `SOC_DROM_LOW` and `SOC_DRAM_HIGH`, for example.